saas/api/roles.py

```diff
@@ -723,13 +722,34 @@ class RoleFilteredListAPIView(RoleSmartListMixin, RoleByDescrQuerysetMixin,
     """
     serializer_class = RoleSerializer
 
+    def get(self, request, *args, **kwargs):
+        response = super(RoleFilteredListAPIView, self).get(
+            request, *args, **kwargs)
+        data = {
+            'data': response.data['results'],
+            'invited_count': self.invited_count,
+            'requested_count': self.requested_count,
+        }
+        response.data['results'] = data
+        return response
+
     def get_queryset(self):
-        queryset = super(RoleFilteredListAPIView, self).get_queryset()
-        role_status = self.request.query_params.get('role_status')
-        if role_status:
-            active = (role_status == 'active')
-            return queryset.filter(grant_key__isnull=active)
-        return queryset
+        base = super(RoleFilteredListAPIView, self).get_queryset()
+        self.requested_count = 0
+        self.invited_count = 0
+        for item in base:
```

Comment:
No!
1/ This code should be in the special purpose decorator/paginator.
2/ Definitely should be a queryset.count() and not loop over the entire results.

Reply:
2) I was thinking about avoiding an extra `COUNT` query here